src/models/db/transfers.js

```diff
@@ -69,42 +71,64 @@ function convertToPersistent (data) {
     data.status_id = transferStatuses.getTransferStatusId(data.state)
     delete data.state
   }
-  data.transfer_id = data.id
+  data.transfer_uuid = data.id
   delete data.id
   return _.mapKeys(data, (value, key) => key.toUpperCase())
 }
 
-function * getTransfer (id, options) {
-  return db.selectOne({TRANSFER_ID: id}, options && options.transaction)
+function * getTransfer (uuid, options) {
+  return db.selectOne({TRANSFER_UUID: uuid}, options && options.transaction)
     .then((transfer) => {
-      return adjustments.getAdjustments(id, options).then((adjustments) => {
+      if (transfer === null) {
+        return null
+      }
+      return adjustments.getAdjustments(transfer._id, options)
+      .then((adjustments) => {
         const result = _.assign({}, transfer, adjustments)
-        return _.isEmpty(result) ? null : result
+        return _.isEmpty(result) ? null : _.omit(result, '_id')
       })
     })
 }
 
 function * updateTransfer (transfer, options) {
-  return db.update(transfer, {TRANSFER_ID: transfer.id},
-    options && options.transaction).then((result) => {
-      return adjustments.upsertAdjustments(transfer, options).then(() => result)
+  const transaction = options && options.transaction
+  return db.update(transfer, {TRANSFER_UUID: transfer.id}, transaction)
+  .then((result) => {
+    return db.selectOne({TRANSFER_UUID: transfer.id}, transaction)
```

Comment:
Yeah, `update` "Resolves the promise / fulfills the callback with the number of affected rows for the query."